library/src/containers/Channels/Operation.tsx

```diff
@@ -72,23 +76,69 @@ export const OperationComponent: React.FunctionComponent<Props> = ({
             </span>
           </h4>
         </header>
+        {operation.summary && (
+          <div className={bemClasses.element(`${className}-description`)}>
+            <Markdown>{operation.summary}</Markdown>
+          </div>
+        )}
         {operation.description && (
           <div className={bemClasses.element(`${className}-description`)}>
             <Markdown>{operation.description}</Markdown>
           </div>
         )}
+        {operation.bindings && (
+          <BindingsComponent
+            bindings={operation.bindings}
+            title={OPERATION_BINDINGS_TEXT}
+          />
+        )}
         <MessagesComponent messages={messages} inChannel={true} />
       </section>
     );
   }
 
   return (
-    <section className={bemClasses.element(className)}>
+    <section
+      className={bemClasses.element(`${className}-oneOf-${payloadType}`)}
+    >
+      <header
+        className={bemClasses.element(
+          `${className}-oneOf-${payloadType}-header`,
+        )}
+      >
+        <h3>
+          {isPublish && isSubscribe ? (
+            <Badge
+              type={
+                payloadType === PayloadType.PUBLISH
+                  ? BadgeType.PUBLISH
+                  : BadgeType.SUBSCRIBE
+              }
+            />
+          ) : null}
+          <span>
+            {payloadType === PayloadType.PUBLISH
+              ? ONE_OF_FOLLOWING_MESSAGES_PUBLISH_SINGLE_TEXT
+              : ONE_OF_FOLLOWING_MESSAGES_SUBSCRIBE_SINGLE_TEXT}
+          </span>
+        </h3>
+      </header>
+      {operation.summary && (
+        <div className={bemClasses.element(`${className}-description`)}>
+          <Markdown>{operation.summary}</Markdown>
+        </div>
+      )}
```

Comment:
I simply reused the existing styles - I think the action to take here is to create separate styles.
 I just wanted the message to be rendered similar to one-of because the previous rendering omitted information. If a channel had both `subscribe` and `publish` operations the result was really confusing. 
Hence I added the help text and the summary if it exists.